model/Metadata/Service/ClassMetadataSearcher.php

```diff
@@ -116,13 +116,9 @@ private function getRelatedProperties(array $result, array $allProperties): arra
 
     private function getMainClassProperties(string $classUri): array
     {
-        $result = $this->executeQuery('_id', $classUri);
+        $results = (array) $this->executeQuery('_id', $classUri);
 
-        if ($result->getTotalCount() === 0) {
-            return [];
-        }
-
-        return current($result);
+        return !empty($results) ? current($results) : [];
```

Comment:
Are we considering both cases in our automated tests? empty results and not empty results?

Reply:
The empty case is covered by a new `testFindAllUsingElasticSearchWithEmptyClassProperties` test included in this merge request. The non-empty case should be covered by pre-existing tests.